integration/integration_test.go

```diff
@@ -0,0 +1,64 @@
+package integration
+
+import (
+	"testing"
+
+	"sigs.k8s.io/krew/integration/krew"
+)
+
+const (
+	// validPlugin is a valid plugin with a small download size
+	validPlugin = "konfig"
+)
+
+func TestInstall(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+
+	krewTest, cleanup := krew.NewKrewTest(t)
+	defer cleanup()
+
+	krewTest.WithIndex().Cmd("install", validPlugin).RunOrFail()
+}
+
+func TestUpdate(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+
+	krewTest, cleanup := krew.NewKrewTest(t)
+	defer cleanup()
+
+	krewTest.WithIndex().Cmd("update").RunOrFail()
+
+	indexFiles, err := krewTest.TempDir().List("index")
+	if err != nil {
+		t.Error(err)
+	}
+
+	if len(indexFiles) == 0 {
+		t.Error("expected some index files but found none")
+	}
+}
+
+func TestUninstall(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+
+	krewTest, cleanup := krew.NewKrewTest(t)
+	defer cleanup()
+
+	krewTest.WithIndex().Cmd("install", validPlugin).RunOrFailOutput()
+	krewTest.Cmd("remove", validPlugin).RunOrFailOutput()
+
+	indexFiles, err := krewTest.TempDir().List("store")
```

Comment:
I think this is fundamentally testing the wrong thing. It’s testing the implementation detail and not the user contract. 

Once you uninstall a plugin these should happen and we should test them instead.
1. Kubectl-foo no longer in PATH
2. Krew list shows it as uninstalled
3. Krew info fails

Reply:
You are totally right here.